### PR TITLE
[SYCL][NFC] Unify setting kernel arguments

### PR DESCRIPTION
**Pull request by AlexeySachkov**

### CODE REVIEW

**Comment by AlexeySachkov** [comment]
@Ruyk, @Alexander-Johnston, could you please help to verify that this patch works fine with CUDA backend?

File: sycl/source/detail/scheduler/commands.cpp [hunk 1]
```diff
@@ -1639,16 +1639,14 @@ cl_int ExecCGCommand::enqueueImp() {
...
-        pi_mem MemArg = (pi_mem)AllocaCmd->getMemAllocation();
-        Plugin.call<PiApiKind::piextKernelSetArgMemObj>(Kernel, Arg.MIndex, &MemArg);
-#else
         RT::PiMem MemArg = (RT::PiMem)AllocaCmd->getMemAllocation();
+        if (!RT::useBackend(pi::Backend::SYCL_BE_PI_OPENCL)) {
+          Plugin.call<PiApiKind::piextKernelSetArgMemObj>(Kernel, Arg.MIndex,
+                                                          &MemArg);
+          break;
+        }
         Plugin.call<PiApiKind::piKernelSetArg>(Kernel, Arg.MIndex,
```
Comment (Ruyk) [inline sycl/source/detail/scheduler/commands.cpp]: I think an else would be a bit clearer than having two breaks, took me a couple of seconds to understand the change on the diff.

Comment (AlexeySachkov) [reply sycl/source/detail/scheduler/commands.cpp]: Refactored, thanks

File: sycl/source/detail/scheduler/commands.cpp [hunk 2]
```diff
@@ -1639,16 +1639,14 @@ cl_int ExecCGCommand::enqueueImp() {
...
-        pi_mem MemArg = (pi_mem)AllocaCmd->getMemAllocation();
-        Plugin.call<PiApiKind::piextKernelSetArgMemObj>(Kernel, Arg.MIndex, &MemArg);
-#else
         RT::PiMem MemArg = (RT::PiMem)AllocaCmd->getMemAllocation();
-        Plugin.call<PiApiKind::piKernelSetArg>(Kernel, Arg.MIndex,
-                                               sizeof(RT::PiMem), &MemArg);
-        Plugin.call<PiApiKind::piKernelSetArg>(Kernel, Arg.MIndex,
-                                               sizeof(RT::PiMem), &MemArg);
-#endif
+        if (!RT::useBackend(pi::Backend::SYCL_BE_PI_OPENCL)) {
```
Comment (romanovvlad) [inline sycl/source/detail/scheduler/commands.cpp]: Could you please reverse if else blocks so if checks for the positive condition?

Comment (AlexeySachkov) [reply sycl/source/detail/scheduler/commands.cpp]: Done

Comment (romanovvlad) [reply sycl/source/detail/scheduler/commands.cpp]: Thanks!

**Comment by romanovvlad** [comment]
@smaslov-intel ping

**Pull request merged**